### ISSUE TITLE
An error occurs when 'robolectric.offline' of systemProperty is enabled in version 4.9

### ISSUE THREAD

**Issue by Popbrain** [robolectric/robolectric#7868]
### Description

When testing API 33, an error occurs when 'robolectric.offline' is enabled in Version 4.9 .
'robolectric.offline' and 'robolectric.dependency.dir' are the option I'm using to speed up CI.
Up to 32 works fine, but only 33 gives the following error.

Is there any other way to enable API 33 testing in offline mode??

### Steps to Reproduce
I reproduced it with the same version as Robolectric's development environment.
- Robolectric version : 4.9
- Kotlin version : 1.7.20
- Gradle version : 7.5.1
- com.android.tools.build:gradle:7.3.1

#### [build.gradle]
```gradle

def robolectricDependenciesFolder = project.rootDir.path + "/.robolectric_deps"

android {
    namespace 'jp.co.unit.test'
    buildToolsVersion '33.0.1'
    compileSdk 33

    defaultConfig {
        minSdk 21
        targetSdk 33

        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        consumerProguardFiles "consumer-rules.pro"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }
    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_11.toString()
    }

    testOptions {
        unitTests {
            returnDefaultValues = true
            includeAndroidResources = true

            all {
                testLogging {
                    showStackTraces = true
                    exceptionFormat = "full"
                }

                // ********** Specifying this option causes the test to fail. ***********
                systemProperty 'robolectric.offline', 'true'
                systemProperty 'robolectric.dependency.dir', robolectricDependenciesFolder
            }
        }
    }
}
```

```
> ls -l ./.robolectric_deps

-rw-r--r--   1 garhira  staff   161M Nov  8 17:50 android-all-instrumented-12.1-robolectric-8229987-i4.jar
-rw-r--r--   1 garhira  staff   170M Nov  8 17:50 android-all-instrumented-13-robolectric-9030017-i4.jar
```

#### SImple test code
```kotlin
@RunWith(RobolectricTestRunner::class)
@Config(
    sdk = [32, 33]
)
class RoboTest {


    @Test
    fun test() {

        val i = 1 + 1
        assertThat(i).isEqualTo(2)
    }
}
```

> 32 is passed, but 33 failed.

Below is the error.
```
java.io.IOException: Negative seek offset
java.lang.RuntimeException: java.io.IOException: Negative seek offset
	at org.robolectric.res.android.FileMap.guessDataOffsets(FileMap.java:269)
	at org.robolectric.res.android.ZipFileRO.OpenArchive(ZipFileRO.java:62)
	at org.robolectric.res.android.CppApkAssets.LoadImpl_measured(CppApkAssets.java:223)
	at org.robolectric.res.android.CppApkAssets.lambda$LoadImpl$0(CppApkAssets.java:206)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.res.android.CppApkAssets.LoadImpl(CppApkAssets.java:203)
	at org.robolectric.res.android.CppApkAssets.Load(CppApkAssets.java:117)
	at org.robolectric.shadows.ShadowArscApkAssets9.nativeLoad(ShadowArscApkAssets9.java:272)
	at org.robolectric.shadows.ShadowArscApkAssets9.nativeLoad(ShadowArscApkAssets9.java:293)
	at android.content.res.ApkAssets.nativeLoad(ApkAssets.java)
	at android.content.res.ApkAssets.__constructor__(ApkAssets.java:296)
	at android.content.res.ApkAssets.<init>(ApkAssets.java:293)
	at android.content.res.ApkAssets.loadFromPath(ApkAssets.java:145)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.robolectric.shadows.ShadowArscApkAssets9$_ApkAssets_$$Reflector37.loadFromPath(Unknown Source)
	at org.robolectric.shadows.ShadowArscApkAssets9.lambda$loadFromPath$3(ShadowArscApkAssets9.java:239)
	at org.robolectric.shadows.ShadowArscApkAssets9.getFromCacheOrLoad(ShadowArscApkAssets9.java:178)
	at org.robolectric.shadows.ShadowArscApkAssets9.loadFromPath(ShadowArscApkAssets9.java:237)
	at android.content.res.ApkAssets.loadFromPath(ApkAssets.java)
	at android.content.res.AssetManager.createSystemAssetsInZygoteLocked(AssetManager.java:249)
	at android.content.res.AssetManager.getSystem(AssetManager.java:276)
	at org.robolectric.shadows.ShadowResources.getSystem(ShadowResources.java:83)
	at android.content.res.Resources.getSystem(Resources.java)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.onCreate(RoboMonitoringInstrumentation.java:70)
	at org.robolectric.android.internal.AndroidTestEnvironment.createInstrumentation(AndroidTestEnvironment.java:571)
	at org.robolectric.android.internal.AndroidTestEnvironment.setUpApplicationState(AndroidTestEnvironment.java:199)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:331)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:278)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Negative seek offset
	at java.base/java.io.RandomAccessFile.seek(RandomAccessFile.java:589)
	at org.robolectric.res.android.FileMap.guessDataOffsets(FileMap.java:227)
	at org.robolectric.res.android.ZipFileRO.OpenArchive(ZipFileRO.java:62)
	at org.robolectric.res.android.CppApkAssets.LoadImpl_measured(CppApkAssets.java:223)
	at org.robolectric.res.android.CppApkAssets.lambda$LoadImpl$0(CppApkAssets.java:206)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.res.android.CppApkAssets.LoadImpl(CppApkAssets.java:203)
	at org.robolectric.res.android.CppApkAssets.Load(CppApkAssets.java:117)
	at org.robolectric.shadows.ShadowArscApkAssets9.nativeLoad(ShadowArscApkAssets9.java:272)
	at org.robolectric.shadows.ShadowArscApkAssets9.nativeLoad(ShadowArscApkAssets9.java:293)
	at android.content.res.ApkAssets.nativeLoad(ApkAssets.java)
	at android.content.res.ApkAssets.$$robo$$android_content_res_ApkAssets$__constructor__(ApkAssets.java:296)
	at android.content.res.ApkAssets.<init>(ApkAssets.java:293)
	at android.content.res.ApkAssets.$$robo$$android_content_res_ApkAssets$loadFromPath(ApkAssets.java:145)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.robolectric.shadows.ShadowArscApkAssets9$_ApkAssets_$$Reflector37.loadFromPath(Unknown Source)
	at org.robolectric.shadows.ShadowArscApkAssets9.lambda$loadFromPath$3(ShadowArscApkAssets9.java:239)
	at org.robolectric.shadows.ShadowArscApkAssets9.getFromCacheOrLoad(ShadowArscApkAssets9.java:178)
	at org.robolectric.shadows.ShadowArscApkAssets9.loadFromPath(ShadowArscApkAssets9.java:237)
	at android.content.res.ApkAssets.loadFromPath(ApkAssets.java)
	at android.content.res.AssetManager.$$robo$$android_content_res_AssetManager$createSystemAssetsInZygoteLocked(AssetManager.java:249)
	at android.content.res.AssetManager.createSystemAssetsInZygoteLocked(AssetManager.java)
	at android.content.res.AssetManager.$$robo$$android_content_res_AssetManager$getSystem(AssetManager.java:276)
	at android.content.res.AssetManager.getSystem(AssetManager.java)
	... 12 more
```

**Comment by utzcoz** [comment]
@hoisie Does this problem related to https://github.com/robolectric/robolectric/commit/f0fbdce8296ce47e2869adc93f7ed1becf7c44f0?

**Comment by hoisie** [comment]
@utzcoz wow nice catch, it will fix that issue. I didn't realize someone actually ran into this issue. When people use preinstrumented jars (which is the default), this issue does not happen.

I'll CP https://github.com/robolectric/robolectric/commit/f0fbdce8296ce47e2869adc93f7ed1becf7c44f0 into 4.9.1

**Comment by hoisie** [comment]
Should be fixed in 4.9.1
